src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowWalker/Commands/KillProcessCommand.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CmdPal.Ext.WindowWalker.Components;
+using Microsoft.CmdPal.Ext.WindowWalker.Helpers;
+using Microsoft.CmdPal.Ext.WindowWalker.Properties;
+using Microsoft.CmdPal.Extensions;
+using Microsoft.CmdPal.Extensions.Helpers;
+
+namespace Microsoft.CmdPal.Ext.WindowWalker.Commands;
+
+internal sealed partial class KillProcessCommand : InvokableCommand
+{
+    private readonly Window _window;
+
+    public KillProcessCommand(Window window)
+    {
+        Icon = new("\xE74D"); // Delete symbol
+        Name = $"{Resources.wox_plugin_windowwalker_Kill} (Ctrl+Delete)";
+        _window = window;
+    }
+
+    /// <summary>
+    /// Method to initiate killing the process of a window
+    /// </summary>
+    /// <param name="window">Window data</param>
+    /// <returns>True if the PT Run window should close, otherwise false.</returns>
+    private static bool KillProcess(Window window)
+    {
+        // Validate process
+        if (!window.IsWindow || !window.Process.DoesExist || !window.Process.Name.Equals(WindowProcess.GetProcessNameFromProcessID(window.Process.ProcessID), StringComparison.Ordinal))
+        {
+            ExtensionHost.LogMessage(new LogMessage() { Message = $"Can not kill process '{window.Process.Name}' ({window.Process.ProcessID}) of the window '{window.Title}' ({window.Hwnd}), because it doesn't exist." });
+            return false;
+        }
+
+        // Request user confirmation
+        if (SettingsManager.Instance.ConfirmKillProcess)
+        {
+            // TODO GH #138 -- need to figure out how to confirm kill process? should this just be the same status thing... maybe not? Need message box?
```

Comment:
I might actually link #153, cause IMO we should have a nested context for Delete > Confirm/Cancel

(if that makes sense)

Reply:
I'll leave a comment linking to either issue num and we can decide how to address that when we get to those TODOs.